### PR TITLE
Create Searching-Algorithm.md

### LINKED ISSUE #575: [Feature Request] Searching Algorithm
## 🚀 Feature Request
A searching algorithm is a method used to locate a specific element or value within a data structure, such as an array, list, or tree. Searching algorithms can be classified into two main categories: linear and binary searches.

1. Linear Search

Description: This is the simplest search algorithm. It sequentially checks each element in the data structure until it finds the target value or reaches the end of the structure.

Complexity:

Time Complexity: O(n), where n is the number of elements in the array.

Space Complexity: O(1), as it uses a constant amount of extra space.

Use Cases: Useful for unsorted or small datasets.

2. Binary Search

Description: This algorithm is more efficient than linear search but requires the data to be sorted. It repeatedly divides the search interval in half. If the target value is less than the middle element, it continues searching in the left half; if it's greater, it searches the right half.

Complexity:

Time Complexity: O(log n).

Space Complexity: O(1) for the iterative version and O(log n) for the recursive version (due to call stack).

Use Cases: Ideal for large, sorted datasets.

3. Other Searching Algorithms

Depth-First Search (DFS): Used in tree and graph structures, this algorithm explores as far down a branch as possible before backtracking.

Breadth-First Search (BFS): Also for trees and graphs, it explores all neighbor nodes at the present depth before moving on to nodes at the next depth level.

Interpolation Search: An improvement over binary search for uniformly distributed data, it estimates the position of the target value.

Exponential Search: Works on unbounded or infinite lists and finds the range where the target may exist before applying binary search.

Applications

Searching algorithms are fundamental in computer science and are widely used in various applications, including:

Database querying

Information retrieval systems

AI and machine learning

Networking and routing protocols

Choosing the right searching algorithm depends on the specific requirements of the application, such as the size of the dataset, whether it is sorted, and the need for efficiency.

### Is your feature request related to a problem? Please describe.
A clear and concise description of what the problem is. Example: "I'm frustrated when I get question where this algorithm are used by default.

### Describe the solution you'd like
I will include all the searching algorithms.

### Additional context

---

**Would you like to work on this feature?**  
- [ I will work]Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by Bhumika-00**
### Description
Updated searching algo

Fixes #575

### Type of change

- [no ] Bug fix (non-breaking change which fixes an issue)
- [ yes] New feature (non-breaking change which adds functionality)
- [no ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ yes] Documentation update

### Checklist:

- [yes ] My code follows the style guidelines of this project
- [yes ] I have performed a self-review of my code
- [yes ] I have commented my code, particularly in hard-to-understand areas
- [yes ] I have added tests that prove my fix is effective or that my feature works
- [no ] New and existing unit tests pass locally with my changes
- [no ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Bhumika-00** [comment]
I have done this please review it

**Comment by Bhumika-00** [comment]
It's completed

File: arrays/Searching-Algorithm.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        if A[mid] == target then
+            return mid
+        else if A[mid] < target then
+            left = mid + 1
+        else
+            right = mid - 1
+    end while
+    return -1
+end procedure
+
```
Comment (ajay-dhangar) [inline arrays/Searching-Algorithm.md:39]: write your code example like

````md
```
// your code here
```
````

for example 

````md
```
 procedure binarySearch(A : sorted list of items, target)
    left = 0
    right = length(A) - 1
    while left <= right do
        mid = (left + right) / 2
        if A[mid] == target then
            return mid
        else if A[mid] < target then
            left = mid + 1
        else
            right = mid - 1
    end while
    return -1
end procedure
```
````

Comment (ajay-dhangar) [inline arrays/Searching-Algorithm.md:39]: write your code example like

````md
```
// your code here
```
````

for example 

````md
```
 procedure binarySearch(A : sorted list of items, target)
    left = 0
    right = length(A) - 1
    while left <= right do
        mid = (left + right) / 2
        if A[mid] == target then
            return mid
        else if A[mid] < target then
            left = mid + 1
        else
            right = mid - 1
    end while
    return -1
end procedure
```
````

File: arrays/Searching-Algorithm.md [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+##Complexity
+- Time Complexity: O(log n)
```
Comment (ajay-dhangar) [inline arrays/Searching-Algorithm.md:41]: replace `O(log n)` with `$O(log n)$`

File: arrays/Searching-Algorithm.md [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+- Space Complexity: O(1)
```
Comment (ajay-dhangar) [inline arrays/Searching-Algorithm.md:42]: and also `O(1)` with `$O(1)$`

File: arrays/Searching-Algorithm.md [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+  return (
+    <div>
+      <h3>Advanced Searching Algorithms</h3>
+      <p><b>Jump Search Index:</b> {jumpIndex}</p>
+      <p><b>Exponential Search Index:</b> {exponentialIndex}</p>
+      <p><b>Interpolation Search Index:</b> {interpolationIndex}</p>
+    </div>
+  )
+}
+
```
Comment (ajay-dhangar) [inline arrays/Searching-Algorithm.md:218]: write your code example within 

````md
```jsx
// your code here
```
````

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by Bhumika-00** [comment]
Changes have been made into pr #718

**Comment by Bhumika-00** [comment]
Please review pr #718

**Pull request closed without merging**